src/stream-tcp-reassemble.c

```diff
@@ -1983,6 +1983,9 @@ int StreamTcpReassembleHandleSegment(ThreadVars *tv, TcpReassemblyThreadCtx *ra_
         }
     } else if (ssn->state == TCP_CLOSED) {
         dir = UPDATE_DIR_BOTH;
+    } else if ((ssn->flags & STREAMTCP_FLAG_ASYNC) != 0) {
+        dir = UPDATE_DIR_PACKET;
```

Comment:
Is there a SV test for this change ?

Reply:
it makes a SV test pass that was broken by other changes in the PR (direction update stuff)